File: app/src/main/java/com/water/datasample/SqliteActivity.java
package com.water.datasample;

import android.app.AlertDialog;
import android.content.ContentValues;
import android.content.DialogInterface;
import android.database.Cursor;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.AdapterView;
import android.widget.Button;
import android.widget.CursorAdapter;
import android.widget.ListView;
import android.widget.SimpleCursorAdapter;

import com.water.datasample.helper.SQLiteHelper;

public class SqliteActivity extends AppCompatActivity {
    private ListView lvStudents;
    private Button btnAdd;
    private Button btnSearch;
    private SQLiteHelper _sh;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_sqlite);

        bindViews();
    }

    private void bindViews() {
        lvStudents = findViewById(R.id.lvStudents);
        btnAdd = findViewById(R.id.btnAdd);
        btnSearch = findViewById(R.id.btnSearch);

        this.setTitle("浏览学生信息");
        _sh = new SQLiteHelper(this);

        //查询数据，获取游标
        fillListView("0,19");

        //提示对话框
        final AlertDialog.Builder builder=new AlertDialog.Builder(this);

        //设置ListView单击监听器
        lvStudents.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> arg0, View arg1, int arg2, long arg3) {
                final long temp=arg3;

                builder.setMessage("真的要删除该记录吗？").setPositiveButton("是",
                        new DialogInterface.OnClickListener() {
                            @Override
                            public void onClick(DialogInterface dialogInterface, int which) {
                                //删除数据
                                _sh.delete("student", (int) temp);

                                fillListView("0,19");
                            }
                        }).setNegativeButton("否", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int which) {

                    }
                });
                AlertDialog dialog=builder.create();
                dialog.show();
            }
        });

        btnAdd.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View view) {
                try{
                    ContentValues cvs = new ContentValues();
                    cvs.put("name", "Lisi");
                    cvs.put("sex", "男");
                    _sh.insert("student", cvs);

                    long count = _sh.getCount("student");

                    fillListView(String.format("%d,%d",(count - 19) < 0? 0 : count - 19, count));
                }
                catch (Exception ex)
                {
                    Log.i("", ex.getMessage());
                }
            }
        });

        btnSearch.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View view) {
                try{
                    long count = _sh.getCount("student");

                    fillListView(String.format("%d,%d",(count - 19) < 0? 0 : count - 19, count));
                }
                catch (Exception ex)
                {
                    Log.i("", ex.getMessage());
                }
            }
        });
    }
    private void fillListView(String limit){
        String[] columns = {"id as _id","name","sex"};
        // 查询数据，获取游标
        Cursor cursor=_sh.query("student", columns, limit);

        //列表项数据
        String[] from={"_id","name","sex"};

        // 列表项ID
        int[] to ={R.id.tvID, R.id.tvName, R.id.tvSex};

        try {
            // 适配器
            SimpleCursorAdapter adapter = new SimpleCursorAdapter(
                    getApplicationContext(),
                    R.layout.list_view_item, cursor, from, to,
                    CursorAdapter.FLAG_REGISTER_CONTENT_OBSERVER);

            //为列表视图添加适配器
            lvStudents.setAdapter(adapter);
        }
        catch (Exception ex)
        {
            Log.i("", ex.getMessage());
        }
    }
}

File: app/src/main/java/com/github/androidhappyclub/datasample/SqliteActivity.kt
/*
 * MIT License
 *
 * Copyright (c) 2023 AndroidHappyClub
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.github.androidhappyclub.datasample

import android.content.ContentValues
import android.os.Bundle
import android.view.View
import android.widget.AdapterView
import android.widget.AdapterView.OnItemClickListener
import android.widget.CursorAdapter
import android.widget.SimpleCursorAdapter
import androidx.appcompat.app.AppCompatActivity
import com.ave.vastgui.tools.view.dialog.MaterialAlertDialogBuilder
import com.ave.vastgui.tools.viewbinding.viewBinding
import com.github.androidhappyclub.datasample.databinding.ActivitySqliteBinding
import com.github.androidhappyclub.datasample.helper.SQLiteHelper
import com.github.androidhappyclub.datasample.log.mLogFactory

class SqliteActivity : AppCompatActivity(R.layout.activity_sqlite) {

    private val binding by viewBinding(ActivitySqliteBinding::bind)
    private val logger = mLogFactory.getLog(SqliteActivity::class.java)
    private val mDialogBuilder by lazy {
        MaterialAlertDialogBuilder(this)
    }
    private lateinit var sqliteHelper: SQLiteHelper

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        this.title = "浏览学生信息"
        bindViews()
    }

    private fun bindViews() {
        sqliteHelper = SQLiteHelper(this)
        //查询数据，获取游标
        fillListView("0,19")
        //设置ListView单击监听器
        binding.lvStudents.onItemClickListener =
            OnItemClickListener { _: AdapterView<*>?, _: View?, _: Int, position: Long ->
                mDialogBuilder
                    .setMessage("真的要删除该记录吗？")
                    .setPositiveButton("是") { _, _ ->
                        //删除数据
                        sqliteHelper.delete("student", position.toInt())
                        fillListView("0,19")
                    }
                    .setNegativeButton("否", null)
                    .show()
            }

        binding.btnAdd.setOnClickListener {
            try {
                val cvs = ContentValues().apply {
                    put("name", "Lisi")
                    put("sex", "男")
                }
                sqliteHelper.insert("student", cvs)
                val count = sqliteHelper.getCount("student")
                fillListView(String.format("%d,%d", if (count - 19 < 0) 0 else count - 19, count))
            } catch (ex: Exception) {
                logger.i(ex.message.toString())
            }
        }

        binding.btnSearch.setOnClickListener {
            try {
                val count = sqliteHelper.getCount("student")
                fillListView(String.format("%d,%d", (count - 19).coerceAtLeast(0), count))
            } catch (ex: Exception) {
                logger.i(ex.message.toString())
            }
        }
    }

    private fun fillListView(limit: String) {
        val columns = arrayOf("id as _id", "name", "sex")
        // 查询数据，获取游标
        val cursor = sqliteHelper.query("student", columns, limit)
        //列表项数据
        val from = arrayOf("_id", "name", "sex")
        // 列表项ID
        val to = intArrayOf(R.id.tvID, R.id.tvName, R.id.tvSex)
        try {
            // 适配器
            val adapter = SimpleCursorAdapter(this, R.layout.list_view_item, cursor, from, to, CursorAdapter.FLAG_REGISTER_CONTENT_OBSERVER)
            // 为列表视图添加适配器
            binding.lvStudents.adapter = adapter
        } catch (ex: Exception) {
            logger.i(ex.message.toString())
        }
    }
}